src/cts/src/HTreeBuilder.cpp

```diff
@@ -363,103 +346,463 @@ void adjustToplevelTopology(Point<double>& a,
+    Point<double> branchPoint,
+    Point<double> parentPoint,
+    double targetDist,
+    const std::vector<Point<double>>& sinks,
+    int scalingFactor)
+{
+  double px = parentPoint.getX();
+  double py = parentPoint.getY();
+  std::vector<Point<double>> candidates;
+  Point<double> point = branchPoint;
+
+  // try points that are on edges of "Manhattan square"
+  // with the parent point in the center
+  //              p16 p1 p9
+  //            p8         p5
+  //          p15            p10
+  //        p4        pp       p2
+  //         p14             p11
+  //           p7           p6
+  //              p13 p3 p12
+  addCandidatePoint(px, py + targetDist, point, candidates); // p1
+  addCandidatePoint(px + targetDist, py, point, candidates); // p2
+  addCandidatePoint(px, py - targetDist, point, candidates); // p3
+  addCandidatePoint(px - targetDist, py, point, candidates); // p4
+
+  double leng50 = targetDist * 0.5;
+  addCandidatePoint(px + leng50, py + leng50, point, candidates); // p5
+  addCandidatePoint(px + leng50, py - leng50, point, candidates); // p6
+  addCandidatePoint(px - leng50, py - leng50, point, candidates); // p7
+  addCandidatePoint(px - leng50, py + leng50, point, candidates); // p8
+
+  double leng25 = targetDist * 0.25;
+  double leng75 = targetDist * 0.75;
+  addCandidatePoint(px + leng25, py + leng75, point, candidates); // p9
+  addCandidatePoint(px + leng75, py + leng25, point, candidates); // p10
+  addCandidatePoint(px + leng75, py - leng25, point, candidates); // p11
+  addCandidatePoint(px + leng25, py - leng75, point, candidates); // p12
+  addCandidatePoint(px - leng25, py - leng75, point, candidates); // p13
+  addCandidatePoint(px - leng75, py - leng25, point, candidates); // p14
+  addCandidatePoint(px - leng75, py + leng25, point, candidates); // p15
+  addCandidatePoint(px - leng25, py + leng75, point, candidates); // p16
+
+  // check if any corners of Manhanttan square are inside some blockage
+  // if so add candidate points that intersect blockage and Manhattan square
+  // p1 is top corner
+  point.setX(px);
+  point.setY(py + targetDist);
+  addCandidatePointsAlongBlockage(point, parentPoint, targetDist,
+				  scalingFactor, candidates, "top");
+
+  // p2 is right corner
+  point.setX(px + targetDist);
+  point.setY(py);
+  addCandidatePointsAlongBlockage(point, parentPoint, targetDist,
+				  scalingFactor, candidates, "right");
+  
+  // p3 is bottom corner
+  point.setX(px);
+  point.setY(py - targetDist);
+  addCandidatePointsAlongBlockage(point, parentPoint, targetDist,
+				  scalingFactor, candidates, "bottom");
+
+  // p4 is left corner
+  point.setX(px - targetDist);
+  point.setY(py);
+  addCandidatePointsAlongBlockage(point, parentPoint, targetDist,
+				  scalingFactor, candidates, "left");
+
+  // try moving cell along x or y, with some offset
+  double bx = branchPoint.getX();
+  double by = branchPoint.getY();
+  Point<double> noBest(-1e6, -1e6);
```

Comment:
const

Reply:
branchPoint has been declared as const & and replaced -1e6 with std::numeric_limits<double>::min().